### PR TITLE
Updated Loqusdb table to display information on if current case is the one counted when the observations equals one

### PR DESCRIPTION
**Pull request by TereseBo**
This PR adds a functionality and fixes a bug.

The changes in this PR implements a check which determines if the current case is included in the observations of a specific variant from a Loqusdb instance. When observations=1 the information is used to choose which text and tooltip to display (Observed only in this case / Observed only in another case). The text is displayed regardless of if any cases from the [families] registered in Loqusdb are found in the scout-database. 

These changes was intended as a fix for https://github.com/Clinical-Genomics/scout/issues/4384, the changes however does not fully solve the issue. But it closes https://github.com/Clinical-Genomics/scout/issues/3469 since the same-case verification is now implemented when observations=1.

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

**How to test**:
1. Identify suitable test-variants which ideally have one observation in at least one of the Louqusdb instances. One where the case_id from scout is present in the [families] of the Loqusdb observations for the variant and one where it is not.
2. Navigate to the variant page in scout. The tr with information on if the observation is from the same or another case should be visible and the text-content correct. If the observation was made for another case and the user has access to the case it is displayed next to the text. If the case was not found in scout or the user does not have access no link will be present. 
3. Check the tooltips which have been added. Same-case and other-case observations have differing tooltip contents. 
4. The main tooltip for the card referring to greyed out links was moved to the actual missing links.

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] code approved by DN
- [x] tests executed by TB

### CODE REVIEW

**Comment by TereseBo** [comment]
On the main branch when obs.observations =1 and no cases were present in obs.cases 
(url:643594-300M:5532511b412214368e668578f574c913

643594-300M:5532511b412214368e668578f574c913

The corresponding image for this branch (Text displays when obs. cases is empty, which occur when the case_id match the family for which the observation have been made):
Screenshot 2024-03-07 at 11 59 21



On main branch, when obs.observations =1 and a case were present in obs.cases:
(url:643594:4c7d5c70d955875504db72ef8e1abe77)
643594:4c7d5c70d955875504db72ef8e1abe77

Corresponding image for this branch (Text and order of appearance for text and case-link has been switched, and a tooltip included):
Screenshot 2024-03-07 at 11 58 59


The main tooltip for the card on this branch:
Screenshot 2024-03-07 at 11 58 24

And the tooltip for a missing link which is visible on hover of the affected link: 
Screenshot 2024-03-07 at 12 00 11

**Comment by TereseBo** [comment]
I ended up keeping "case" in the text for observations=1 since this seems to be what is actually indicated. Individual is not quite right since the observations seems to calculated on a case basis. When the same individual is run again in a different case (with new id) an identity check might prevent the data from being uploaded to Loqusdb but I it seems this also prevents the case_id from appearing in the [families] for the variant and the observations will then remain as 1 and belonging to a different case. In this instance perhaps using the term "family" will cause more confusion than case since it would imply something other than case is intended?

File: scout/server/blueprints/variant/templates/variant/variant_details.html [hunk 1]
```diff
@@ -165,24 +165,36 @@
...
-                         <span class="ml-3 badge rounded-pill bg-light text-dark">{{ data.case.display_name }}</span>
-                       {% endif %}
-                     {% endfor %}
-                     {% if obs.cases|length > 10 %}
-                       + more
-                     {% endif %}
-                     {% if obs.observations == 1 %}
-                       Observed only in this case
-                     {% endif %}
+                 {% if obs.observations == 1 %}
```
Comment (dnil) [inline scout/server/blueprints/variant/templates/variant/variant_details.html:171]: I have gotten quite used to some indentation in the templates, though it is of course not functional or necessary. It does help visually grouping blocks. 😊

File: scout/server/blueprints/variant/templates/variant/variant_details.html [hunk 2]
```diff
@@ -165,24 +165,36 @@
                <td>{{ obs.homozygote|default('N/A') }}</td>
                <td>{{ obs.total|default('N/A') }}</td>
              </tr>
-           {% if obs.cases %}
+           {% if obs.cases or obs.observations == 1 %}
```
Comment (dnil) [inline scout/server/blueprints/variant/templates/variant/variant_details.html:168]: 👍

File: scout/server/blueprints/variant/templates/variant/variant_details.html [hunk 3]
```diff
@@ -165,24 +165,36 @@
...
-                     {% endfor %}
-                     {% if obs.cases|length > 10 %}
-                       + more
-                     {% endif %}
-                     {% if obs.observations == 1 %}
-                       Observed only in this case
-                     {% endif %}
+                 {% if obs.observations == 1 %}
+                 {% if obs.case_match==True %}
+                 Observed only in this case
```
Comment (dnil) [inline scout/server/blueprints/variant/templates/variant/variant_details.html:173]: Yes, that must be so, must it not! 😁

File: scout/server/blueprints/variant/templates/variant/variant_details.html [hunk 4]
```diff
@@ -165,24 +165,36 @@
...
+                 <span id="case_match_tooltip"  data-bs-toggle='tooltip' title="Subsequent analysis of the same individual or multiple occurrences within the same family will usually not result in an increased observation count, even if performed by different institutes.">
+                   <i class="mx-1 fas fa-info-circle"></i>
+                 </span>
+                 {% elif obs.case_match ==False %}
```
Comment (dnil) [inline scout/server/blueprints/variant/templates/variant/variant_details.html:177]:
```suggestion
                 {% elif obs.case_match == False %}
```
Being consistent with whitespaces around operators is nice, but hey, its not like we always do.

File: scout/server/blueprints/variant/templates/variant/variant_details.html [hunk 5]
```diff
@@ -165,24 +165,36 @@
...
-                       {% endif %}
-                     {% endfor %}
-                     {% if obs.cases|length > 10 %}
-                       + more
-                     {% endif %}
-                     {% if obs.observations == 1 %}
-                       Observed only in this case
-                     {% endif %}
+                 {% if obs.observations == 1 %}
+                 {% if obs.case_match==True %}
```
Comment (dnil) [inline scout/server/blueprints/variant/templates/variant/variant_details.html:172]:
```suggestion
                 {% if obs.case_match == True %}
```

File: scout/server/blueprints/variant/templates/variant/variant_details.html [hunk 6]
```diff
@@ -165,24 +165,36 @@
...
+                 Observed only in this case
+                 <span id="case_match_tooltip"  data-bs-toggle='tooltip' title="Subsequent analysis of the same individual or multiple occurrences within the same family will usually not result in an increased observation count, even if performed by different institutes.">
+                   <i class="mx-1 fas fa-info-circle"></i>
+                 </span>
+                 {% elif obs.case_match ==False %}
+                 Observed in one other case
+                 <span id="case_mismatch_tooltip"  data-bs-toggle='tooltip' title="Low sequencing quality for the variant or previous loqusdb upload of data from the same individuals may prevent the current case from occurring in loqusdb">
```
Comment (dnil) [inline scout/server/blueprints/variant/templates/variant/variant_details.html:179]: 🎖️ I would say this leaves very little to fix in #4384. I think we can close it after merging.

**Review by dnil — APPROVED** [review]
Yes, thank you! This now solves #3469, and between the wording and tooltips, the risk of misunderstanding from #4384 should now be so slim as to be better ignored.

**Pull request merged**